docs/vm-configuration.md

```diff
@@ -264,3 +264,95 @@ the generation of multiple libvirt XML documents, the domain XML and associated
 documents for secrets, network filters and node devices. So there is not a strict
 1-1 mapping between the k8s API resource representation and the libvirt low
 level representation.
+
+Proposed VM Definition
+----------------------
+
+The following definition adheres to main points described above. It provides
+
+1. high level description usable by end users or management applications and
+2. low level details required to construct and launch a VM.
+
+The features section is used to generate the domain section, the other way
+around is out of scope for this proposal. Each feature must represent one or
+more fields in the domain section.
+
+The design and contents of each feature section should be logically split and
+discussed in separate proposal. This is just an example without the sections
+discussed, converted from cluster/vm.yaml. Splitting the features enables us to
+create modular VM Spec generation flows that are easily tunable by experts in
+their domains of expertise.
+
+```yaml
+apiVersion: kubevirt.io/v1alpha2
+kind: VM
+metadata:
+  name: testvm
+spec:
+  features:
```

Comment:
Yes, `VMConfig` was intended to be a higher-level representation and a way to store non-running VMs.

A thought is to think about:

```yaml
kind: VMConfig
spec:
  features:
    .. # as described in this proposal
  domain:
    .. # to override stuff as needed

# THIS IS THEN TRANSFORMED on /start to a VM OBJ

kind: VM
spec:
  domain:
    ..
```

Thus `features` would just be a property of the `VMConfig` object, and be translated during start into the domain.